### PR TITLE
glvisualize backend

### PR DESCRIPTION
**Pull request by SimonDanisch**
Work in progress for the GLVisualize backend

### CODE REVIEW

File: src/backends.jl [hunk 1]
```diff
@@ -26,18 +26,18 @@ macro init_backend(s)
     end)
 end
 
-@init_backend Immerse
-@init_backend Gadfly
```
Comment (tbreloff) [inline src/backends.jl]: Is there a reason you commented these out?

File: src/examples.jl [hunk 2]
```diff
@@ -252,7 +252,7 @@ PlotExample("DataFrames",
         iris = RDatasets.dataset("datasets", "iris")
         scatter(iris, :SepalLength, :SepalWidth, group=:Species,
             title = "My awesome plot", xlabel = "Length", ylabel = "Width",
-            marker = (0.5, [:+ :h :star7], 12), bg=RGB(.2,.2,.2))
```
Comment (tbreloff) [inline src/examples.jl]: Can you revert this?

File: src/output.jl [hunk 3]
```diff
@@ -210,7 +211,7 @@ end
 # IJulia
 # ---------------------------------------------------------
 
-const _ijulia_output = String["text/html"]
```
Comment (tbreloff) [inline src/output.jl]: Please revert this

Comment (SimonDanisch) [inline src/output.jl]: yeah of course, that's why it's a "do not merge" ;) These edits are just there to make my life a bit easier temporarily!

**Comment by SimonDanisch** [comment]
- [ ] correct handling of all the different color attributs
- [ ] text and labels
- [ ] a few 3D plots still don't work perfectly
- [ ] better camera for corner cases (extreme zoom values etc)
- [ ] shape up shape plots
- [ ] fix efficient renderloop (currently inefficient again, since there are weird artifacts otherwise)
- [ ] fix line anti aliasing for all cases
- [ ] make wireframe work for (x,y,z) surface plots

**Comment by SimonDanisch** [comment]
okay this starts looking okay!
Anyone daring to try this out should be on GLWindow, GLAbstraction, GLVisualize, FixedSizeArrays and GeometryTypes master!
New issues:
- [ ] 2D camera can now rotate which is extremely annoying :D Restrict movement!
- [ ] outline of sprites + barplots still not satisfactory
- [ ] 3D axis
- [ ] annotations
- [ ] Reactive message queue fills up with long blocking loops without yield (e.g. @gif)

Here's a little (badly compressed) gif:
surf

It's embarrassing slow though, since GLVisualize and Plots build up the complete plot for every frame which is completely unnecessary! ;) Will need to figure out a API/hijack the existing one, to bypass this!

**Comment by SimonDanisch** [comment]
can someone try this out?! :)

**Comment by SimonDanisch** [comment]
gotta be on:
GLWindow, GLAbstraction, GLVisualize and GeometryTypes master and for FixedSizeArrays on:
https://github.com/SimonDanisch/FixedSizeArrays.jl/pull/157

**Comment by SimonDanisch** [comment]
Should we merge this?
And @tbreloff, could you add me as a collaborator, so that I can develop in a branch? It'll be easier to check out new versions of the backend like this!

**Comment by tbreloff** [comment]
We should, but I'm traveling right now. Did you rebase on a recent dev?

**Comment by tbreloff** [comment]
What a historic day... :tada: :tada: :tada:

File: src/axes.jl [hunk 4]
```diff
@@ -309,15 +309,13 @@ function expand_extrema!(sp::Subplot, d::KW)
 
     # expand for bar_width
     if d[:seriestype] == :bar
-        dsym = vert ? :x : :y
+        dsym = (!vert) ? :x : :y
```
Comment (tbreloff) [inline src/axes.jl]: I just reverted this change... any idea why you did this?  The idea is that it's setting a default bar width and expanding the axis extrema.  When the plot is vertical, we want to update the x axis.

Comment (SimonDanisch) [inline src/axes.jl]: Well, it was skewed in the wrong direction so I changed this and then it was working...

**Pull request merged**